ipam/ring/ring.go

```diff
@@ -374,26 +393,36 @@ func (r *Ring) ClaimForPeers(peers []mesh.PeerName) {
 	defer r.assertInvariants()
 	defer r.updateExportedVariables()
 
-	totalSize := r.distance(r.Start, r.End)
-	share := totalSize/address.Count(len(peers)) + 1
-	remainder := totalSize % address.Count(len(peers))
-	pos := r.Start
+	r.createEntries(peers)
+	r.Seeds = peers
+}
 
-	for i, peer := range peers {
-		if address.Count(i) == remainder {
-			share--
-			if share == 0 {
-				break
-			}
+// createEntries subdivides the [from,to) (CIDR) range for the given peers
+// and creates entries for the subranges.
+// Each entry is CIRD-aligned.
+func (r *Ring) createEntries(peers []mesh.PeerName) {
```

Comment:
I'd prefer to have a separate function to not overbloat the `ClaimForPeers` method.